src/global-shortcuts.c

```diff
@@ -566,18 +566,85 @@ handle_list_shortcuts (XdpDbusGlobalShortcuts *object,
   return G_DBUS_METHOD_INVOCATION_HANDLED;
 }
 
+static void
+shortcuts_configure_cb (GObject *source_object,
+                        GAsyncResult *res,
+                        gpointer data)
+{
+  GDBusMethodInvocation *invocation = data;
+  g_autoptr(GError) error = NULL;
+
+  if (!xdp_dbus_impl_global_shortcuts_call_configure_shortcuts_finish (impl, res, &error))
+    {
+      g_dbus_error_strip_remote_error (error);
+      g_warning ("Failed to configure shortcuts: %s", error->message);
+      g_dbus_method_invocation_return_gerror(invocation, error);
+      return;
+    }
+
+  xdp_dbus_global_shortcuts_complete_configure_shortcuts(XDP_DBUS_GLOBAL_SHORTCUTS(global_shortcuts), invocation);
+}
+
+static XdpOptionKey global_shortcuts_configure_shortcuts_options[] = {
+  { "activation_token", G_VARIANT_TYPE_STRING, NULL },
+};
+
+static gboolean
+handle_configure_shortcuts (XdpDbusGlobalShortcuts *object,
+                            GDBusMethodInvocation *invocation,
+                            const gchar *arg_session_handle,
+                            const gchar *arg_parent_window,
+                            GVariant *arg_options)
+{
+  XdpCall *call = xdp_call_from_invocation (invocation);
+  XdpSession *session;
+  g_autoptr(GError) error = NULL;
+  g_auto(GVariantBuilder) options_builder =
+    G_VARIANT_BUILDER_INIT (G_VARIANT_TYPE_VARDICT);
+
+
+  if (!xdp_filter_options (arg_options, &options_builder,
+                           global_shortcuts_configure_shortcuts_options,
+                           G_N_ELEMENTS (global_shortcuts_configure_shortcuts_options),
+                           &error))
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return G_DBUS_METHOD_INVOCATION_HANDLED;
+    }
+
+  session = xdp_session_from_call (arg_session_handle, call);
+  if (!session)
+    {
+      g_dbus_method_invocation_return_error (invocation,
+                                             G_DBUS_ERROR,
+                                             G_DBUS_ERROR_ACCESS_DENIED,
+                                             "Invalid session");
+      return G_DBUS_METHOD_INVOCATION_HANDLED;
+    }
+
+  xdp_dbus_impl_global_shortcuts_call_configure_shortcuts (impl,
+                                                           arg_session_handle,
+                                                           arg_parent_window,
+                                                           g_variant_builder_end (&options_builder),
+                                                           NULL,
+                                                           shortcuts_configure_cb,
+                                                           invocation);
```

Comment:
You have to keep the invocation alive, so `g_object_ref` here, and assign to an autoptr (or otherwise unref at the end) in `shortcuts_configure_cb`.